neuraxle/hyperparams/space.py

```diff
@@ -226,6 +227,61 @@ class HyperparameterSamples(RecursiveDict):
     def __init__(self, *args, separator=None, **kwds):
         super().__init__(*args, separator=separator, **kwds)
 
+    def compress(self, trim_parents=False):
+        """Compresses the HyperparameterSamples representation """
+        grouped_result = self._group_hps_by_step(flat_steps_hps=self.items())
+        return self._convert_to_compressed_format(group_by_step=grouped_result, trim_parents=trim_parents)
```

Comment:
These functions could be somewhere else and do not belong to the present class. Here is a suggestion: 
```suggestion
    return CompressedHyperparams.from(self)
```

Reply:
Okay, I am thinking to move both `_group_hps_by_step` and `_convert_to_compressed_format` methods under `CompressedHyperparamsSamples` class. So `HyperparamsSamples.compress()` will only return `CompressedHyperparamsSamples` class object directly. Is it what expected? @guillaume-chevalier